paper/src/main/java/vg/civcraft/mc/civmodcore/world/locations/chunkmeta/WorldChunkMetaManager.java

```diff
@@ -41,8 +44,7 @@ public class WorldChunkMetaManager {
 	 */
 	private final Set<ChunkCoord> unloadingQueue;
 	private final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
-	private Thread chunkLoadingConsumer;
-	private LinkedBlockingQueue<ChunkCoord> chunkLoadingQueue;
+	private final Executor chunkLoadingExecutor = Executors.newCachedThreadPool();
```

Comment:
This is an unbounded threadpool, meaning it will create threads on demand if no thread is free to do work. So this may create 1000 threads if necessary, taking up 1000 connections in the process.